### PR TITLE
Fixes for changeling transformations.

### PR DESCRIPTION
**Pull request by Christmas5**
## About The Pull Request

One of the main issues with shapeshifting was that _/mutant_bodyparts_ and _/body_markings_ in _/datums/dna_ did not update. This PR should fix that by adding these two lines to _/proc/transform_ in _changeling.dm:_
```
user.dna.mutant_bodyparts = chosen_dna.mutant_bodyparts.Copy()
user.dna.body_markings = chosen_dna.body_markings.Copy()
```

This PR also adds a bunch of stuff such as underwear colours and eye emissives to the changeling profiles, allowing these to get copied over when shapeshifting, thus making the disguise more accurate and/or harder to see through.

**ISSUES LEFT:**
- Body size does not update when using lesser form. Shapeshift into someone with a body size/scale of 1 before using lesser form, otherwise you risk permanently messing up your sprite for the round. (Same thing applies to last resort, your body size has to be exactly 1.0 before you use it.)
- Clothes do not update properly when using lesser form/human form, take them on/off to fix it.
- I've had eye colours break with emissive eyes when using lesser form.
- When using last resort, sometimes the injected egg won't hatch, effectively roundremoving the changeling.
- /tg/'s default lizard tails do not copy over when shapeshifting, not an issue if the target has ticked "Tail" in their prefs.
- Haven't tested this with characters that can change their form at will, f.e. slimes.
- I also didn't fully test the changes with vox and tesh characters.

## Changelog

**Changeling profiles save more information about the targets appearance.**
- Colours for underwear, undershirts and socks.
- Gradient styles and gradient colours for hair and facial hair.
- Eye emissives.
- Physique.
- Neck items.
- Paths to digi/vox/tesh variants of worn icons as well as variation flags for flesh disguise items.

**Changeling transformations run _/transfer_identity(), /updateappearance()_ and _/regenerate_icons()_ one more time at the end.**
Currently in order for a transformation to finish, the ability has to be used twice. The necessary code will now run **twice** when using the ability **once.**

**Variation flags and variation icons for flesh disguise items are now saved in changeling profiles.**
Flesh disguise items did not always save the digi/vox/tesh icons for clothes, they also did not update their variation flags. Both the icons and flags are now saved in changeling profiles, allowing flesh disguises to use digi/vox/tesh icons when appropriate.

**/updateappearance() updates the eye colour of /organ/eyes if eyecolor_update is set to  1.**
Characters have _eye_color_left/right_ which update correctly, but the eyes themselves **also** have these variables. This addition allows _/updateappearance()_ to update _eye_color_left/right_ in _/organ/eyes_ **on top of** updating the character's _eye_color_left/right._

**Lesser form will now clear out mutant parts, body markings and flesh disguises.**
_/mutant_bodyparts, /body_markings_ and flesh disguises should now clear out before transforming the character into a monkey, preventing them from becoming an abomination.

**Last resort will now respawn you as a /human rather than /human/species/monkey.**
As the egg hatches, the game will spawn a human and turn them into a monkey, using code similar to that of lesser form (instead of spawning an actual monkey). This should prevent issues with randomised names.

:cl:
fix: Changeling transformations should finish after using the ability once instead of twice.
fix: Changeling transformations now update mutant parts, body markings, eye colours, eye emissives, underwear/undershirt/socks colours, gradient styles, gradient colours, physique, neck items.
/:cl:

### CODE REVIEW

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Multi-line modularization comments work like
```
//SKYRAT EDIT START
code
more code
//SKYRAT EDIT END

**Comment by Christmas5** [comment]
> Multi-line modularization comments work like
> 
> ```
> //SKYRAT EDIT START
> code
> more code
> //SKYRAT EDIT END
> ```

My bad, I think it looks alright now?

**Review by Christmas5 — COMMENTED** [review]
Fixed(?)

File: code/modules/antagonists/changeling/changeling.dm [hunk 1]
```diff
@@ -482,7 +495,8 @@
 	// Grab the target's sechut icon.
 	new_profile.id_icon = target.wear_id?.get_sechud_job_icon_state()
 
-	var/list/slots = list("head", "wear_mask", "back", "wear_suit", "w_uniform", "shoes", "belt", "gloves", "glasses", "ears", "wear_id", "s_store")
+	//var/list/slots = list("head", "wear_mask", "back", "wear_suit", "w_uniform", "shoes", "belt", "gloves", "glasses", "ears", "wear_id", "s_store")
```
Comment (Zonespace27) [inline code/modules/antagonists/changeling/changeling.dm]: remove commented code

File: code/modules/antagonists/changeling/changeling.dm [hunk 2]
```diff
@@ -665,7 +686,8 @@
 	var/static/list/slot2slot = list(
 		"head" = ITEM_SLOT_HEAD,
 		"wear_mask" = ITEM_SLOT_MASK,
-		"neck" = ITEM_SLOT_NECK,
+		//"neck" = ITEM_SLOT_NECK,
```
Comment (Zonespace27) [inline code/modules/antagonists/changeling/changeling.dm]: ditto

File: code/modules/antagonists/changeling/changeling.dm [hunk 3]
```diff
@@ -683,14 +705,27 @@
...
+	// SKYRAT EDIT END
 
 	chosen_dna.transfer_identity(user, TRUE)
 
 	for(var/obj/item/bodypart/limb as anything in user.bodyparts)
 		if(IS_ORGANIC_LIMB(limb))
 			limb.update_limb(is_creating = TRUE)
 
-	user.updateappearance(mutcolor_update = TRUE)
+	//user.updateappearance(mutcolor_update = TRUE)
```
Comment (Zonespace27) [inline code/modules/antagonists/changeling/changeling.dm]: ditto

File: code/modules/antagonists/changeling/powers/humanform.dm [hunk 4]
```diff
@@ -29,10 +29,19 @@
 	changeling.purchased_powers -= src
 	Remove(user)
 
-	var/newmob = user.humanize()
+	//var/newmob = user.humanize()
```
Comment (Zonespace27) [inline code/modules/antagonists/changeling/powers/humanform.dm]: ditto

File: code/modules/antagonists/changeling/powers/lesserform.dm [hunk 5]
```diff
@@ -14,12 +14,32 @@
...
 	var/datum/antagonist/changeling/changeling = user.mind.has_antag_datum(/datum/antagonist/changeling)
 	var/datum/action/changeling/humanform/from_monkey/human_form_ability = new()
 	changeling.purchased_powers += human_form_ability
 	changeling.purchased_powers -= src
+	
+	// SKYRAT EDIT START
+	for(var/slot in changeling.slot2type)
+		if(istype(user.vars[slot], changeling.slot2type[slot]))
+			qdel(user.vars[slot])
+	for(var/i in user.all_scars)
```
Comment (Zonespace27) [inline code/modules/antagonists/changeling/powers/lesserform.dm]: no one letter vars

File: code/modules/mob/living/simple_animal/hostile/headcrab.dm [hunk 6]
```diff
@@ -83,9 +83,17 @@
 		if(changeling_datum.can_absorb_dna(owner))
 			changeling_datum.add_new_profile(owner)
 
-		var/datum/action/changeling/humanform/hf = new
+		//var/datum/action/changeling/humanform/hf = new
+		//changeling_datum.purchased_powers += hf
```
Comment (Zonespace27) [inline code/modules/mob/living/simple_animal/hostile/headcrab.dm]: remove commented code

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Pull request merged**